app/src/main/java/com/kks/nimblesurveyjetpackcompose/ui/presentation/home/HomeScreen.kt

```diff
@@ -2,15 +2,255 @@ package com.kks.nimblesurveyjetpackcompose.ui.presentation.home
+    val swipeableState = rememberSwipeableState(initialValue = "M")
+    val sizePx = LocalDensity.current.run { LocalConfiguration.current.screenWidthDp.toDp().toPx() }
+    // Maps anchor points (in px) to states
+    val anchors = mapOf(0f to "L", sizePx / 2 to "M", sizePx to "R")
+    var threshold by remember { mutableStateOf(IDLE) }
+    val isRefreshing by viewModel.isRefreshing.collectAsState()
+
+    SwipeRefresh(
+        state = rememberSwipeRefreshState(isRefreshing),
+        onRefresh = { },
+        modifier = Modifier.fillMaxSize()
+    ) {
+        LazyColumn {
+            if (isRefreshing && surveyList.isEmpty()) {
+                item { HomeScreenShimmerLoading(modifier = Modifier.fillParentMaxSize()) }
+            }
+            if (surveyList.isNotEmpty() && error == null) {
+                item {
+                    LaunchedEffect(keys = arrayOf(swipeableState.offset.value), block = {
+                        val currentSwipeState = swipeableState.offset.value
+                        if (currentSwipeState < sizePx / 2 && threshold == IDLE) {
+                            // Swipe to left
+                            threshold = LEFT_SWIPE
+                        } else if (currentSwipeState > sizePx / 2 && threshold == IDLE) {
+                            // Swipe to right
+                            threshold = RIGHT_SWIPE
+                        } else if ((currentSwipeState == sizePx / 2)) {
+                            if (
+                                (selectedSurveyNumber < surveyList.size - 1 && threshold == LEFT_SWIPE) ||
+                                (selectedSurveyNumber != 0 && threshold == RIGHT_SWIPE)
+                            ) {
+                                selectedSurveyNumber += threshold
+                            }
+                            threshold = IDLE
+                        }
+                        swipeableState.snapTo(targetValue = "M")
+                    })
+                    SurveyContent(
+                        modifier = Modifier
+                            .fillParentMaxSize()
+                            .swipeable(
+                                state = swipeableState,
+                                anchors = anchors,
+                                thresholds = { _, _ -> FractionalThreshold(fraction = FRACTION) },
+                                orientation = Orientation.Horizontal
+                            ),
+                        numberOfPage = surveyList.size,
+                        selectedSurveyNumber = selectedSurveyNumber,
+                        surveyResponse = surveyList[selectedSurveyNumber]
+                    )
+                }
+            }
+        }
+    }
+
     BackHandler {
         activity?.finish()
     }
 }
+
+@Suppress("DestructuringDeclarationWithTooManyEntries")
+@Composable
+fun SurveyContent(modifier: Modifier, surveyResponse: SurveyResponse, numberOfPage: Int, selectedSurveyNumber: Int) {
```

Comment:
```suggestion
fun SurveyContent(
    modifier: Modifier, 
    surveyResponse: SurveyResponse, 
    numberOfPage: Int, 
    selectedSurveyNumber: Int
) {
```

Reply:
It is in one line with `hard wrap at: 120`. I thought you wanted me to set it to `120` at [this comment](https://github.com/kaungkhantsoe/NimbleSurveyJetpackCompose/pull/72#discussion_r954496599) before no?

Reply:
That's true, but in the [compass code conventions](https://nimblehq.co/compass/development/code-conventions/kotlin/#formatting), we also add line breaks when there are more than 2 parameters in a method 😉

Reply:
I see 👌